client/components/form-sections/application-table.tsx

```diff
@@ -1,141 +1,171 @@
-  };
-  const tdStyles =
-    'table-td px-6 py-4 text-left text-sm font-strong flexitems-center justify-between';
-  return (
-    <tbody>
-      {applications.applications &&
-        applications.applications.map((row: any, index: number) => (
-          <tr
-            key={`row${index}`}
-            onClick={() => handleSelectRow(row.BCAAP_Form_ID)}
-            className='bg-white border-b-2 even:bg-bcGrayInput
-              border-gray-200'
-          >
-            <td className={tdStyles}>{row.Confirmation_ID}</td>
-            <td className={tdStyles}>{row.BCAAP_Form_ID}</td>
-            <td className={tdStyles}>{row.Assigned_To}</td>
-            <td className={tdStyles}>{row.Created_At}</td>
-            <td className={tdStyles}>{row.Status}</td>
-            <td className={tdStyles}>{row.Assigned_To}</td>
-            <td className={tdStyles}>{row.Created_At}</td>
-            <td className={tdStyles}>{row.Status}</td>
-          </tr>
-        ))}
-    </tbody>
-  );
-};
+export const ApplicationDashboard: React.FC<any> = () => {
+  const [state, setState] = useState({
+    searchFacilityName: '',
+    searchConfirmatioID: '',
+    totalApplications: 0,
+    data: [],
+  });
 
-const ApplicationTable: React.FC<Props> = applications => {
-  return (
-    <div>
-      <table className='min-w-full text-center'>
-        <TableHeader />
-        {ApplicationTableData && applications.applications.length != 0 ? (
-          <TableBody applications={applications.applications} />
-        ) : (
-          <p className='text-center text-sm mt-4'>No applications found.</p>
-        )}
-      </table>
-    </div>
-  );
-};
+  const { searchFacilityName, searchConfirmatioID, totalApplications, data } = state;
 
-export const ApplicationDashboard: React.FC<any> = () => {
-  const [currentPage, setCurrentPage] = useState<number>(1);
-  const [applicationsPerPage] = useState<number>(10);
-  const [searchValue, setSearchValue]: [string, (search: string) => void] = useState('');
+  const { push, query } = useRouter();
+  const { fetchData } = useHttp();
 
-  const handleSearch = (e: { target: { value: string } }) => {
-    setSearchValue(e.target.value);
+  const setApplicationData = async (params: any) => {
+    try {
+      fetchData(
+        {
+          endpoint: '/applications',
+          params,
+        },
+        ({ result, total }: any) => {
+          setState(state => ({ ...state, data: result, totalApplications: total }));
+        },
+      );
+    } catch (err) {
+      // console.log('Error occured when fetching applications');
+    }
   };
 
-  // Filter data with search value
-  const filteredApplications =
-    ApplicationTableData &&
-    ApplicationTableData.filter((item: any) => {
-      return item.Confirmation_ID.toLowerCase().includes(searchValue.toLowerCase());
-    });
+  useEffect(() => {
+    (async () => {
+      push(
+        { query: { ...query, page: 1, limit: 1, facilityName: '', confirmationId: '' } },
```

Comment:
Shouldn't replace the values on load. 1st should check if page, limit, facilityName and confirmationId is set. If not should update these values!